### PR TITLE
@benisgold/memo helpers

### PR DESCRIPTION
**Pull request by benisgold**
This PR applies `memoFn` from https://github.com/rainbow-me/rainbow/pull/2637 to relevant functions within src/helpers.

### CODE REVIEW

**Comment by skylarbarrera** [comment]
looks like you got some linting

File: src/utils/memoFn.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export function memoFn<TArgs extends unknown[], TReturn extends unknown>(
       }
     }
 
-    const key = `key ${args.join(' ~ ')}`;
```
Comment (terrysahaidak) [inline src/utils/memoFn.ts:62]: Please, rever this change

**Review by terrysahaidak — CHANGES_REQUESTED** [review]
Looks good overall except for that small change

**Review by terrysahaidak — APPROVED** [review]

**Review by skylarbarrera — APPROVED** [review]
send it

**Pull request merged**